web/src/pages/api/public/ingestion.ts

```diff
@@ -212,6 +212,36 @@ export default async function handler(
 
     const sortedBatch = sortBatch(batch);
 
+    // We group events by eventBodyId which allows us to store and process them
+    // as one which reduces infra interactions per event. Only used in the S3 case.
+    const sortedBatchByEventBodyId = sortedBatch.reduce(
+      (
+        acc: Record<
+          string,
+          {
+            data: IngestionEventType[];
+            key: string;
+            type: (typeof eventTypes)[keyof typeof eventTypes];
+          }
+        >,
+        event,
+      ) => {
+        if (!event.body?.id) {
+          return acc;
+        }
```

Comment:
If there is no body or id, we skip the upload and worker processing. Based on the typing this should only occur for SdkLog events.